src/tools/continuousintegration.rst

```diff
@@ -79,6 +79,101 @@ the button “Run pipeline” on a specific branch (i.e. master).
 
 |image5|
 
+Collecting & consolidating CI health metrics as part of the CI pipeline
+-----------------------------------------------------------------------
+As part of the CI/CD process all teams are expected to collect and consolidate
+the necessary code health metrics under a file named :code:`ci-metrics.json` at
+the root of the repositories they have ownership over.
+
+The :code:`ci-metrics.json` file is expect to be created automatically as part
+of the CI pipeline by the teams by collecting the relevant information from the
+*unit tests*, *coverage*, *linting* and *build status*.
+**An important point to notice, is that :code:`ci-metrics.json` shouldn't exist
+as part of the repository, but, be created specifically as part of the CI
+pipeline.**
+The file must be created and properly populated before the start of the marked
+:code:`# START: Gitlab CI badges creation` in the **pages** stage of the CI
+pipeline (:code:`.gitlab-ci.yml` file).
+
+The metrics should be collected under the following structure:
+
+- build-status: *top level placeholder for the build process status*
+
+  - last: *placeholder about the last build process*
+
+    - status: (**string**) *failed or passed, stating the last build status*
+    - timestamp: (**float**) *the Unix timestamp with the date and time of the
```

Comment:
can we allow timestamps to be string representations according to a defined format as well as unix timestamps?  maybe using something like the _dateutil_ module?

Reply:
The badge is created with a text defined format from the timestamp in the file. So the info in the badge (provided automatically by the badge creation utility we provide) will be in the format `2019/10/11 12:45:12`.

A unix time timestamp is the standard way of storing this kind of information. I think that duplicating the same information in the same file in a different format is not needed and it can induce in error.